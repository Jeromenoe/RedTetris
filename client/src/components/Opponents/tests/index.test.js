import React from 'react';
import { Opponents } from '../index';
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import { Provider } from 'react-redux';



describe('Testing Opponents Component', () => {
	test('Single Opponent', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				opponents: [
					{
						name: "toto",
						spectrum: [
							[0, 0, 'I', 'J']
						]
					}
				]
			}
		};
		const store = mockStore(initialState);
		const tree = renderer.create(<Provider store={store}><Opponents /></Provider>).toJSON();
		expect(tree).toMatchSnapshot();
	});
	test('multiple opponents', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				opponents: [
					{
						name: "toto",
						spectrum: [
							[0, 0, 'I', 'J']
						]
					},
					{
						name: "titi",
						spectrum: [
							[0, 0, 'I', 'J']
						]
					}
				]
			}
		};
		const store = mockStore(initialState);
		const tree = renderer.create(<Provider store={store}><Opponents /></Provider>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('Opponent spectrum height = 0', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				opponents: [
					{
						name: "toto",
						spectrum: []
					}
				]
			}
		};
		const store = mockStore(initialState);
		const tree = renderer.create(<Provider store={store}><Opponents /></Provider>).toJSON();
		expect(tree).toMatchSnapshot();
	});
});