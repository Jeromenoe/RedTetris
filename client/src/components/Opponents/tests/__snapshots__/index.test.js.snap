// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Opponents Component Opponent spectrum height = 0 1`] = `
Array [
  <div
    style={
      Object {
        "background": "rgba(0,0,0,0.7)",
        "height": "100%",
        "width": "1px",
      }
    }
  />,
  <aside>
    <div
      className="sc-gtsrHT gQOXp"
    >
      <div
        className="sc-dlnjwi bBrhSr"
      >
        <div
          className="sc-bdnxRM kwkGEb"
          height={0}
          width={0}
        />
         
        <br />
         
        toto
      </div>
    </div>
  </aside>,
]
`;

exports[`Testing Opponents Component Single Opponent 1`] = `
Array [
  <div
    style={
      Object {
        "background": "rgba(0,0,0,0.7)",
        "height": "100%",
        "width": "1px",
      }
    }
  />,
  <aside>
    <div
      className="sc-gtsrHT gQOXp"
    >
      <div
        className="sc-dlnjwi bBrhSr"
      >
        <div
          className="sc-bdnxRM eHxUYu"
          height={1}
          width={4}
        >
          <div
            className="sc-hKFxyN ccLPrQ"
            color="ecf0f1"
            type="0"
          />
          <div
            className="sc-hKFxyN ccLPrQ"
            color="ecf0f1"
            type="0"
          />
          <div
            className="sc-hKFxyN ecIMNg"
            color="148f77"
            type="spectrum"
          />
          <div
            className="sc-hKFxyN ecIMNg"
            color="148f77"
            type="spectrum"
          />
        </div>
         
        <br />
         
        toto
      </div>
    </div>
  </aside>,
]
`;

exports[`Testing Opponents Component multiple opponents 1`] = `
Array [
  <div
    style={
      Object {
        "background": "rgba(0,0,0,0.7)",
        "height": "100%",
        "width": "1px",
      }
    }
  />,
  <aside>
    <div
      className="sc-gtsrHT gQOXp"
    >
      <div
        className="sc-dlnjwi bBrhSr"
      >
        <div
          className="sc-bdnxRM eHxUYu"
          height={1}
          width={4}
        >
          <div
            className="sc-hKFxyN ccLPrQ"
            color="ecf0f1"
            type="0"
          />
          <div
            className="sc-hKFxyN ccLPrQ"
            color="ecf0f1"
            type="0"
          />
          <div
            className="sc-hKFxyN ecIMNg"
            color="148f77"
            type="spectrum"
          />
          <div
            className="sc-hKFxyN ecIMNg"
            color="148f77"
            type="spectrum"
          />
        </div>
         
        <br />
         
        toto
      </div>
      <div
        className="sc-dlnjwi bBrhSr"
      >
        <div
          className="sc-bdnxRM eHxUYu"
          height={1}
          width={4}
        >
          <div
            className="sc-hKFxyN ccLPrQ"
            color="ecf0f1"
            type="0"
          />
          <div
            className="sc-hKFxyN ccLPrQ"
            color="ecf0f1"
            type="0"
          />
          <div
            className="sc-hKFxyN ecIMNg"
            color="148f77"
            type="spectrum"
          />
          <div
            className="sc-hKFxyN ecIMNg"
            color="148f77"
            type="spectrum"
          />
        </div>
         
        <br />
         
        titi
      </div>
    </div>
  </aside>,
]
`;
